### DISCUSSION TITLE
dns flooding

### DISCUSSION THREAD

**Discussion by gtxaspec** [gtxaspec/wz_mini_hacks#114] [General]
does anyone still experience this issue?  dns flooding/spamming on their networks

**Comment by virmaior** [comment]
right now I've got mine back on the internet so I haven't been able to check.

**Comment by gtxaspec** [comment]
right now, i've got a test v3 camera setup, i installed dnsmasq on it, and i see dns queries to c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com about every 5 seconds.

```
Jun 21 04:00:42 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is NODATA-IPv6
Jun 21 04:00:42 dnsmasq[5182]: query[A] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:00:42 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 52.88.0.147
Jun 21 04:00:42 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 35.155.238.14
Jun 21 04:00:42 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 54.185.42.140
Jun 21 04:00:47 dnsmasq[5182]: query[AAAA] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:00:47 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is NODATA-IPv6
Jun 21 04:00:47 dnsmasq[5182]: query[A] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:00:47 dnsmasq[5182]: forwarded c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com to 8.8.8.8
Jun 21 04:00:47 dnsmasq[5182]: forwarded c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com to 8.8.4.4
Jun 21 04:00:47 dnsmasq[5182]: reply c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 35.155.238.14
Jun 21 04:00:47 dnsmasq[5182]: reply c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 52.88.0.147
Jun 21 04:00:47 dnsmasq[5182]: reply c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 54.185.42.140
Jun 21 04:00:53 dnsmasq[5182]: query[AAAA] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:00:53 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is NODATA-IPv6
Jun 21 04:00:53 dnsmasq[5182]: query[A] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:00:53 dnsmasq[5182]: forwarded c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com to 8.8.8.8
Jun 21 04:00:53 dnsmasq[5182]: reply c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 35.155.238.14
Jun 21 04:00:53 dnsmasq[5182]: reply c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 52.88.0.147
Jun 21 04:00:53 dnsmasq[5182]: reply c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 54.185.42.140
Jun 21 04:00:58 dnsmasq[5182]: query[AAAA] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:00:58 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is NODATA-IPv6
Jun 21 04:00:58 dnsmasq[5182]: query[A] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:00:58 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 54.185.42.140
Jun 21 04:00:58 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 52.88.0.147
Jun 21 04:00:58 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 35.155.238.14
Jun 21 04:01:04 dnsmasq[5182]: query[AAAA] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:01:04 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is NODATA-IPv6
Jun 21 04:01:04 dnsmasq[5182]: query[A] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:01:04 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 35.155.238.14
Jun 21 04:01:04 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 54.185.42.140
Jun 21 04:01:04 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 52.88.0.147
Jun 21 04:01:09 dnsmasq[5182]: query[AAAA] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:01:09 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is NODATA-IPv6
Jun 21 04:01:09 dnsmasq[5182]: query[A] c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com from 127.0.0.1
Jun 21 04:01:09 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 52.88.0.147
Jun 21 04:01:09 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 35.155.238.14
Jun 21 04:01:09 dnsmasq[5182]: cached c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com is 54.185.42.140
```

**Reply by sideup66** [reply]
maybe thisis something we can add on startup, also check for traffic on port 10001, that is weirdly enough what it floods out on, according to this doc here:
https://www.speedguide.net/port.php?port=10001

that port is used by wyze for p2p live streaming, yet I see nothing using it on mine...im curious what would happen to the camera if i just flat out block that port on outbound.

**Reply by gtxaspec** [reply]
do you have any tcpdump logs from port 10001?  I have no traffic on that port, even with P2P streaming active.

**Reply by sideup66** [reply]
Nothing at this moment, can quickly capture one, though i am experimenting with that port as mine always has a few unreplied connections out on it.

**Comment by gtxaspec** [comment]
A solution I have found to stop the dns queries for certain:

`touch /tmp/test_hosts;echo -e "127.0.0.1 localhost\n54.185.42.140 c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com" > /tmp/test_hosts; mount --bind /tmp/test_hosts /etc/hosts`

this stops them, the query never makes it to the dns server.

**Reply by sideup66** [reply]
Sorry for delay, here is the pcap :) note I had a stream running over 5g interestingly enough as this was streaming...
[out.pcap.zip](https://github.com/gtxaspec/wz_mini_hacks/files/8945871/out.pcap.zip)
.

**Reply by sideup66** [reply]
Im also wondering how I could potentially make a script to setup this dns entry, this way if the IP ever changes the camera would just quietly detect that on boot and be able to update accordingly.

**Comment by clee604** [comment]
Is there a way to make this solution more permanent? I find this doesn't survive reboots and even without reboots it randomly stops working and has to be reapplied.

**Reply by gtxaspec** [reply]
yes, i will add a function to wz_mini.conf to prevent dns flooding, its a planned feature

**Comment by gtxaspec** [comment]
latest release has the config option `ENABLE_LOCAL_DNS`

set this to true to enable a local caching dns server, no more flooding to your network.

**Reply by clee604** [reply]
thank you!! I'll upgrade my cameras and test it out asap

**Reply by gtxaspec** [reply]
any feedback? =D

**Reply by clee604** [reply]
So far no news is good news! 

I upgraded all my cameras yesterday (both V3 and V2) and so far everything is working fine. Pihole also doesn't report the AWS DNS calls so that's good.

Great work!

**Comment by Vendo232** [comment]
1. just ran update script and the ENABLE_LOCAL_DNS option is not there ( V2 ) but dnsmasq.conf is present now in /etc/
2. How do you guys use the DNSMASQ on your local net? Should I point to my pihole IP? 

some usecase would be great how to setup the dnsmasq.conf

![image](https://user-images.githubusercontent.com/44098230/180851768-a43fdd87-7f7d-4af9-b524-3927d59e07ac.png)

**Reply by clee604** [reply]
upgrading doesn't change your conf file, so just add the line as per the template conf file here:
https://github.com/gtxaspec/wz_mini_hacks/blob/master/SD_ROOT/wz_mini/wz_mini.conf

My cameras' DNS server is set via DHCP on my router, so it points to my pihole there so I can see the DNS queries.

**Reply by Vendo232** [reply]
thx, I have the same config, I see/block DNS queries in pihole. Does it mean that this add-on has no use for us?
![image](https://user-images.githubusercontent.com/44098230/180859950-4c063f52-b8ac-426e-847c-54f664dd4af0.png)

was not sure if this DNSMASQ could bring a permanent ( restart resistant ) block for the "c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com" ping

**Reply by clee604** [reply]
This DNS parameter prevents the actual query from being made.. reducing the load on your pihole server and the network. I find when the camera is flooding with these DNS packets it has some performance impacts to the video stream as well.

**Reply by Vendo232** [reply]
this sounds great! stopping the ping should help with performance.

I would like to learn how you setup the DNS on camera to achieve it.

**Reply by clee604** [reply]
you mean how to turn it on? Or how this works technically?

to turn it on, it's a parameter you can add to your wz_mini.conf file
ENABLE_LOCAL_DNS="true"

**Reply by gtxaspec** [reply]
right, @Vendo232 just add that to your config on the latest wz_mini version. 

Make sure you also disable any previous dns related stuff you were doing in any scripts, as to not interfere.  

If you want to use your pihole's upstream dns, just edit `wz_mini/etc/resolv.dnsmasq`, delete the default ones and leave only your pihole's ip, `nameserver 192.168.1.3` for example

no other configuration is required.

**Reply by clee604** [reply]
ah i didn't realize the nameservers are hardcoded now in the dnsmasq file. I'll update mine to point to my pihole to confirm the DNS flooding stopped

**Reply by gtxaspec** [reply]
any flooding should hit dnsmasq and not go out of the camera

**Reply by clee604** [reply]
true, but if the nameserver isn't pointing to the pihole I have no way of confirming. Unfortunately I don't have a USG or any other type of packet identification service

**Comment by clee604** [comment]
after updating the resolv.dnsmasq to point to my pihole, I can see there are still DNS queries happenening to the AWS domain (c1ybkrkbr1j10x.credentials.iot.us-west-2.amazonaws.com).

They don't seem as regular as every 5 seconds now, but they still occur. This happens on one of the cameras more than the others

One other camera also pinging google and wyze a lot which didn't happen before...

![image](https://user-images.githubusercontent.com/93049965/181028689-52e56c23-af2c-407d-ae5f-090792e11618.png)

So far I haven't been able to figure out why the camera behaviour isn't consistent.

**Reply by gtxaspec** [reply]
@clee604 

try adding:

`min-cache-ttl=3600`

to the bottom of `dnsmasq.conf` and see if this decreases the queries on your pihole

**Reply by gtxaspec** [reply]
I also just released a new update to wz_mini, i patched dnsmasq to allow min-cache-ttl values up to 86400 (1 day), so it will only ask for the domain name to the upstream server, once a day.  

Without the new version, the max can only be set up to 3600 (1 hour)

**Reply by clee604** [reply]
i noticed the wz_mini/etc/resolv.dnsmasq file does not survive upgrade (at least this morning's update). I have edit the nameserver again.. i was wondering why the cameras went silent all day on the pihole

**Reply by gtxaspec** [reply]
all files inside the wz_mini folder, except for etc/configs and etc/ssh and etc/rc.local.d are replaced with the package versions. just fyi

**Reply by clee604** [reply]
as a future feature request, can we have some parameters in the wz_mini.conf file to put in custom nameservers?

**Comment by gtxaspec** [comment]
@Vendo232 the latest version supports the hosts file `wz_mini/etc/hosts` if you are using `ENABLE_LOCAL_DNS` in addition to the dns flood protection.  You can place domains here to block, for example. `127.0.0.1 whatever.com` and those lookups will be rerouted to localhost (blocked)
